### PR TITLE
[HB-6535] InMobi bid API support

### PR DESCRIPTION
**Pull request by bwised**
https://chartboost.atlassian.net/browse/HB-6535

Bid support with InMobi, SDK 10.7.0.

Their documentation:
[Bidding Integration with InMobi SDK.pdf](https://github.com/ChartBoost/chartboost-mediation-ios-adapter-inmobi/files/15298693/Bidding.Integration.with.InMobi.SDK.pdf)

### CODE REVIEW

File: Source/InMobiAdapter.swift [hunk 1]
```diff
@@ -65,8 +77,13 @@ final class InMobiAdapter: NSObject, PartnerAdapter {
     /// - parameter request: Information about the ad load request.
     /// - parameter completion: Closure to be performed with the fetched info.
     func fetchBidderInformation(request: PreBidRequest, completion: @escaping ([String : String]?) -> Void) {
-        // InMobi does not currently provide any bidding token
-        completion(nil)
+        log(.fetchBidderInfoStarted(request))
+        guard let bidToken = IMSdk.getToken() else {
+            log(.fetchBidderInfoFailed(request, error: InMobiAdapterError.failedToProvideBidToken))
```
Comment (daniel-barros) [inline Source/InMobiAdapter.swift:82]: we could simplify removing the need for a custom error.
```suggestion
            log(.fetchBidderInfoFailed(request, error: error(.prebidFailureUnknown, description: "Failed to provide bid token.")))
```

**Review by daniel-barros — CHANGES_REQUESTED** [review]
Don't we need to update the load logic to take into account the adm for bidding placements?
That's generally the way other partner SDKs work.

**Comment by bwised** [comment]
Indeed.  I am missing that part from their docs.

**Comment by bwised** [comment]
I addressed it.  I also attached their documentation to the description if anybody needs to look at it.

File: Source/InMobiAdapter.swift [hunk 2]
```diff
@@ -65,8 +66,13 @@ final class InMobiAdapter: NSObject, PartnerAdapter {
     /// - parameter request: Information about the ad load request.
     /// - parameter completion: Closure to be performed with the fetched info.
     func fetchBidderInformation(request: PreBidRequest, completion: @escaping ([String : String]?) -> Void) {
-        // InMobi does not currently provide any bidding token
-        completion(nil)
+        log(.fetchBidderInfoStarted(request))
+        guard let bidToken = IMSdk.getToken() else {
+            log(.fetchBidderInfoFailed(request, error: error(.prebidFailureUnknown, description: "Failed to provide bid token.")))
+            return
```
Comment (daniel-barros) [inline Source/InMobiAdapter.swift:72]: Let's call `completion(nil)` here to make the operation as finished.

Comment (bwised) [reply Source/InMobiAdapter.swift:73]: https://github.com/ChartBoost/chartboost-mediation-ios-adapter-inmobi/pull/59/commits/401336ce21222c11b52734a3d085b01a7dcc58aa

**Review by daniel-barros — APPROVED** [review]
LGTM

**Pull request merged**